script.js

```diff
@@ -1 +1,519 @@
+
+      // set the click event
+      // eslint-disable-next-line
+      square.addEventListener('click', (event) => {
+
+        // we will want to pass in the card element so
+        // that we can change how it looks on screen, i.e.,
+        // "turn the card over"
+        squareClick(event.currentTarget, i, j);
+
+        startGameTimer(countDownTime);
+      });
+
+      rowElement.appendChild(square);
+    }
+    boardElement.appendChild(rowElement);
+  }
+
+  return boardElement;
+};
+
+// Function that creates the userName input;
+const getUserNameInput = () => {
+// Creating the input box & submit button to take user name
+  const inputBox = document.createElement('input');
+  inputBox.setAttribute('placeholder', 'Please enter your name!');
+  inputBox.setAttribute('id', 'submit-username');
+
+  const submitButton = document.createElement('button');
+  submitButton.innerHTML = 'Submit';
+
+  const inputDiv = document.createElement('div');
+  // Append inputBox & Submit Button into inputDiv Tag
+  inputDiv.appendChild(inputBox);
+  inputDiv.appendChild(submitButton);
+
+  // Add event listener to submit button to get username
+  submitButton.addEventListener('click', () => {
+    const userName = document.querySelector('#submit-username');
+    outputDivTag.innerHTML = `Welcome ${userName.value}! Please click on a card and the timer will begin.`;
+    inputBox.value = '';
+    console.log('test');
+  });
+  document.body.appendChild(inputDiv);
+};
+// Fn to outputCardDisplay by each nested div inside square divs
+
+// Function that initializes the game;
+const gameInit = () => {
+  // create username and submit button on top first;
+  getUserNameInput();
+  // create this special deck by getting the doubled cards and
+  // making a smaller array that is ( boardSize squared ) number of cards
+  const doubleDeck = makeDeck();
+  const deckSubset = doubleDeck.slice(0, boardSize * boardSize);
+  deck = shuffleCards(deckSubset);
+
+  // deal the cards out to the board data structure
+  for (let i = 0; i < boardSize; i += 1) {
+    board.push([]);
+    positionArray.push([]);
+    for (let j = 0; j < boardSize; j += 1) {
+      board[i].push(deck.pop());
+      positionArray[i].push('');
+    }
+  }
+
+  const boardEl = buildBoardElements(board);
+
+  // Initialize welcome message and append output div and score div
+  outputDivTag.innerHTML = outputValue;
+  scoreOutputDivTag.innerHTML = `Your score is ${score}`;
+
+  // Append all necessary elements below:
+
+  document.body.appendChild(boardEl);
+  document.body.appendChild(outputDivTag);
+  document.body.appendChild(scoreOutputDivTag);
+  document.body.appendChild(timer);
+  document.body.appendChild(resetButton);
+};
+
+const noClickResetGame = () => {
+  // remove all elements under document.body
+  document.body.innerHTML = '';
+  timer.innerHTML = '';
+  clearInterval(ref);
+  restartGlobalVariables();
+  gameInit();
+  resetGame();
+};
+
+const resetGame = () => {
```

Comment:
This function name is misleading. It's called `resetGame`, but it doesn't reset the game. It creates a button that when clicked will reset the game. It's confusing that `resetGame` is called directly below `gameInit` on page load, because what I expect to happen is game initialisation then game reset, which is not what's happening.

It's equally confusing that there's a `noClickResetGame` function, that only runs when the Reset Game button is clicked lol.

Instead of having `resetGame` run at the highest level scope, I would rename `resetGame` to something like `createResetGameButton` and call that function from `gameInit`. I would update the logic in `noClickResetGame` to not run `gameInit` again, but just reset the globals and clear HTML in HTML elements that should be reset.

Thanks for the hard work in thinking through this logic and getting it to work!